pkg/cmd/util/crypto/crypto.go

```diff
@@ -0,0 +1,429 @@
+package crypto
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/golang/glog"
+)
+
+type TLSConfig struct {
+	// Certificate config to serve the master API
+	CA *TLSCertificateConfig
+	// Certificate config to serve the master API
+	Master *TLSCertificateConfig
+	// Certificate config to serve the assets
+	Asset *TLSCertificateConfig
+	// Certificate config to connect to kubernetes
+	KubeClient *TLSCertificateConfig
+	// Certificate config to connect to openshift
+	OpenShiftClient *TLSCertificateConfig
+	// Certificate config for registry to use
+	RegistryClient *TLSCertificateConfig
+	// Certificate config for deployer to use
+	DeployerClient *TLSCertificateConfig
```

Comment:
It feels weird to have certain special users auth information baked into a struct like this.  Is this temporary or permanent?